Open Judge System/Workers/OJS.Workers.ExecutionStrategies/CSharpUnitTestsRunnerExecutionStrategy.cs

```diff
@@ -0,0 +1,226 @@
+    using OJS.Workers.Executors;
+
+    public class CSharpUnitTestsRunnerExecutionStrategy : ExecutionStrategy
+    {
+        private const string ZippedSubmissionName = "Submission.zip";
+        private const string TestedCode = "TestedCode.cs";
+        private const string CsProjFileSearchPattern = "*.csproj";
+        private const string ProjFileExtenstion = ".csproj";
+        private const string DllFileExtension = ".dll";
+
+        public CSharpUnitTestsRunnerExecutionStrategy(string nUnitConsoleRunnerPath)
+        {
+            this.NUnitConsoleRunnerPath = nUnitConsoleRunnerPath;
+            this.WorkingDirectory = DirectoryHelpers.CreateTempDirectory();
+
+        }
+
+        ~CSharpUnitTestsRunnerExecutionStrategy()
+        {
+            DirectoryHelpers.SafeDeleteDirectory(this.WorkingDirectory, true);
+        }
+
+        protected string NUnitConsoleRunnerPath { get; set; }
+
+        protected string WorkingDirectory { get; set; }
+
+        public override ExecutionResult Execute(ExecutionContext executionContext)
+        {
+            ExecutionResult result = new ExecutionResult();
+            byte[] userSubmissionContent = executionContext.FileContent;
+
+            var submissionFilePath = $"{this.WorkingDirectory}\\{ZippedSubmissionName}";
+            File.WriteAllBytes(submissionFilePath, userSubmissionContent);
+            FileHelpers.UnzipFile(submissionFilePath, this.WorkingDirectory);
+            File.Delete(submissionFilePath);
+
+            string csProjFilePath = FileHelpers.FindFirstFileMatchingPattern(
+                this.WorkingDirectory,
+                CsProjFileSearchPattern);
+
+            // In order to run unit tests written for the MSTest framework via the NUnit framework we need to
+            // switch the standard "using Microsoft.VisualStudio.TestTools.UnitTesting" directives to the NUnit ones 
+            // reference: http://www.adamtuliper.com/2009/05/blog-post.html
+            // this.SwitchUsingDirectivesInCsFiles();
+
+            // Edit References in Project file
+            var project = new Project(csProjFilePath);
+            this.CorrectProjectReferences(project);
+            project.Save(csProjFilePath);
+            project.ProjectCollection.UnloadAllProjects();
+
+            // Initially set isCompiledSucessfully to true
+            result.IsCompiledSuccessfully = true;
+
+            var executor = new RestrictedProcessExecutor();
+            var checker = Checker.CreateChecker(
+                executionContext.CheckerAssemblyName,
+                executionContext.CheckerTypeName,
+                executionContext.CheckerParameter);
+
+            result = this.RunUnitTests(executionContext, executor, checker, result, csProjFilePath);
+            return result;
+        }
+
+        private ExecutionResult RunUnitTests(ExecutionContext executionContext, IExecutor executor, IChecker checker, ExecutionResult result, string csProjFilePath)
```

Comment:
Place each parameter on separate line in order to not exceed 120 characters convention limit.